dotCMS/src/main/java/com/dotmarketing/beans/MultiTree.java

```diff
@@ -1,126 +1,176 @@
         return this.treeOrder;
     }
 
-    public void setTreeOrder(int treeOrder) {
+    public MultiTree setTreeOrder(int treeOrder) {
+        treeOrder = (treeOrder<0)? 0  : treeOrder;
         this.treeOrder = treeOrder;
+        return new MultiTree(this.htmlPage, this.container, this.contentlet, this.relationType, treeOrder);
     }
 
     public String toString() {
         return ToStringBuilder.reflectionToString(this);
     }
 
     public boolean equals(Object other) {
-        if ( !(other instanceof MultiTree) ) return false;
+        if (!(other instanceof MultiTree))
+            return false;
         MultiTree castOther = (MultiTree) other;
-        return new EqualsBuilder()
-            .append(this.parent1, castOther.parent1)
-            .append(this.parent2, castOther.parent2)
-            .append(this.child, castOther.child)
-            .isEquals();
+        return new EqualsBuilder().append(this.htmlPage, castOther.htmlPage).append(this.container, castOther.container)
+                .append(this.contentlet, castOther.contentlet).isEquals();
     }
 
     public int hashCode() {
-        return new HashCodeBuilder()
-            .append(parent1)
-            .append(parent2)
-            .append(child)
-            .toHashCode();
-    }
-	/**
-	 * @return Returns the parent1.
-	 */
-	public String getParent1() {
-		return parent1;
-	}
-	/**
-	 * @param parent1 The parent1 to set.
-	 */
-	public void setParent1(String parent1) {
-		this.parent1 = parent1;
-	}
-	/**
-	 * @return Returns the parent2.
-	 */
-	public String getParent2() {
-		return parent2;
-	}
-	/**
-	 * @param parent2 The parent2 to set.
-	 */
-	public void setParent2(String parent2) {
-		this.parent2 = parent2;
-	}
-}
\ No newline at end of file
+        return new HashCodeBuilder().append(htmlPage).append(container).append(contentlet).toHashCode();
+    }
+
+    /**
+     * 
+     * {@link #getHtmlPage()}
+     */
+    @Deprecated
+    public String getParent1() {
+        return htmlPage;
+    }
+    /**
+     * 
+     * {@link #getContainer()}
+     */
+    @Deprecated
+    public String getParent2() {
+        return container;
+    }
+
+    /**
+     * @return Returns the htmlPage.
+     */
+    public String getHtmlPage() {
+        return htmlPage;
+    }
+    /**
+     * 
+     * {@link #setHtmlPage(String)}
+     */
+    @Deprecated
+    public MultiTree setParent1(String htmlPage) {
+        return setHtmlPage(htmlPage);
+    }
+    /**
+     * 
+     * {@link #setContainer(String)}
+     */
+    @Deprecated
+    public MultiTree setParent2(String container) {
+        return setContainer(container);
+    }
+
+    /**
+     * @param htmlPage The htmlPage to set.
+     */
+    public MultiTree setHtmlPage(String htmlPage) {
+        return new MultiTree(htmlPage, this.container, this.contentlet, this.relationType, this.treeOrder);
+    }
+
+    /**
+     * @return Returns the container.
+     */
+    public String getContainer() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "getParent2" on line 125. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)